time stamp
int32 data1
int32 data2
string message
